### PR TITLE
BUG: Fix duplicate plugins on HTML editor fields

### PR DESCRIPTION
**Pull request by lukereative**
Fixes silverstripe/silverstripe-admin#635

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -47,13 +44,40 @@ const config = [
...
     }
   },
+  {
+    name: 'tinymce',
+    entry: {
+      TinyMCE_sslink: `${PATHS.LEGACY_SRC}/TinyMCE_sslink.js`,
+      'TinyMCE_sslink-external': `${PATHS.LEGACY_SRC}/TinyMCE_sslink-external.js`,
+      'TinyMCE_sslink-email': `${PATHS.LEGACY_SRC}/TinyMCE_sslink-email.js`,
+    },
+    output: {
```
Comment (unclecheese) [inline webpack.config.js]: Let's reduce the duplication here. Everything from this point down can be in a common object. Then, for both configs, use `...commonConfig`. For the vendor bundle, use

```js
...commonConfig,
plugins: [
  ...commonConfig.plugins,
  new webpack.optimise.CommonChunks(....
]
```

File: webpack.config.js [hunk 2]
```diff
@@ -47,13 +44,40 @@ const config = [
       // bundles aren't duplicating libs.
       new webpack.optimize.CommonsChunkPlugin({
         name: 'vendor',
-        minChunks: module => module.context && module.context.indexOf('/node_modules/') > -1,
+        minChunks(module) {
```
[the remark this answers is not in the document]
Comment (lukereative) [reply webpack.config.js]: Ah, I thought I'd removed it.

**Comment by unclecheese** [comment]
@lukereative Can you confirm that these API changes to `TinyMCEActionRegistrar` are backward compatible? e.g. If I have a custom action I've registered in my 4.2 project, is it going to work in 4.3 without intervention?

**Comment by lukereative** [comment]
@unclecheese I might actually make a change so if you had a plugin that doesn't have an config identifier, `getActions` will return the actions for the specific config and grab any that were registered without a config identifier, so they aren't completely ignored

File: client/src/lib/TinyMCEActionRegistrar.js [hunk 3]
```diff
@@ -18,26 +28,60 @@ class TinyMCEActionRegistrar {
...
+    return actions;
   }
 
-  getSortedActions(menu) {
-    const actions = this.getActions(menu);
+  /**
+   * Get list of actions for this menu, sorted by first by priority and then text
+   *
+   * @param {String} menu
+   * @param {Bool} configId Identifier of which config
```
Comment (robbieaverill) [inline client/src/lib/TinyMCEActionRegistrar.js]: This isn't a bool. Also would be good to be consistent with the types (Bool vs Boolean), and also note that I think JSDoc generally uses lowercase for scalar types, other than Object and Array which seem to be used in both uppercased and lowercase in examples

**Pull request merged**